src/snowflake/cli/plugins/snowpark/commands.py

```diff
@@ -308,31 +313,96 @@ def _deploy_single_object(
     }
 
 
+deprecated_pypi_download_option = typer.Option(
+    YesNoAsk.NO.value,
+    "--pypi-download",
+    help="Whether to download non-Anaconda packages from PyPi.",
+    hidden=True,
+    callback=deprecated_flag_callback_enum(
+        "--pypi-download flag is deprecated. Snowpark build command"
+        " always tries to download non-Anaconda packages from PyPi."
+    ),
+)
+
+
+def _write_requirements_file(file_path: SecurePath, requirements: List[Requirement]):
+    log.info("Writing requirements into file %s", file_path.path)
+    file_path.write_text("\n".join(req.line for req in requirements))
+
+
 @app.command("build")
 @with_project_definition("snowpark")
 def build(
-    pypi_download: YesNoAsk = PyPiDownloadOption,
-    check_anaconda_for_pypi_deps: bool = CheckAnacondaForPyPiDependencies,
-    package_native_libraries: YesNoAsk = PackageNativeLibrariesOption,
+    ignore_anaconda: bool = IgnoreAnacondaOption,
+    allow_shared_libraries: bool = AllowSharedLibrariesOption,
+    index_url: Optional[str] = IndexUrlOption,
+    skip_version_check: bool = SkipVersionCheckOption,
+    deprecated_package_native_libraries: YesNoAsk = deprecated_allow_native_libraries_option(
+        "--package-native-libraries"
+    ),
+    deprecated_check_anaconda_for_pypi_deps: bool = DeprecatedCheckAnacondaForPyPiDependencies,
+    _deprecated_pypi_download: YesNoAsk = deprecated_pypi_download_option,
     **options,
 ) -> CommandResult:
     """
     Builds the Snowpark project as a `.zip` archive that can be used by `deploy` command.
     The archive is built using only the `src` directory specified in the project file.
     """
-    paths = SnowparkPackagePaths.for_snowpark_project(
+    if not deprecated_check_anaconda_for_pypi_deps:
+        ignore_anaconda = True
+    # TODO: yes/no/ask logic should be removed in 3.0
+    allow_shared_libraries_yesnoask = {
+        True: YesNoAsk.YES,
+        False: YesNoAsk.NO,
+    }[allow_shared_libraries]
+    if deprecated_package_native_libraries != YesNoAsk.NO:
+        allow_shared_libraries_yesnoask = deprecated_package_native_libraries
```

Comment:
I don't like the fact that we want to get rid of this variable but we still use it. I suggested this in #924 but I think splitting the creation step from checking for native libraries allows us to avoid pushing this yesnoask-thing deepper in logic.

1. Create package
2. Check for native code
3. Do validation logic

something like

```py
download_packages()
if packages_has_native_code():
   if not allow_shared_libraries or deprecated_package_native_libraries == YesNoAsk.NO:
       raise ... / return ...
   if deprecated_package_native_libraries == YesNoAsk.ASK:
       prompt(...)
      
package_code()
package_libs()
```

WDYT?

Reply:
Done!
More pros: now error messages are command-dependent